### ISSUE TITLE
[Question] How to insert text into HTML text editor

### ISSUE THREAD

**Issue by eatonjl** [microsoft/playwright#17361]
Hello! I am trying to find out how I can code (in Typescript) the ability to enter text into an HTML text editor. I tried using Codegen but it doesn't recognize me entering text into the HTML text editor. Has anyone had success in doing this?

**Comment by mxschmitt** [comment]
It fully depends on which text editor you are using, they all work differently! Generally speaking using the Keyboard API should work well: https://playwright.dev/docs/api/class-keyboard

Which one are you using? What is not working?

**Comment by eatonjl** [comment]
@mxschmitt Thank you for your response! I think we are using the [ASPxHtmlEditor Class](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxHtmlEditor.ASPxHtmlEditor) in our code. If I select the text box while using codegen the action is not recognized. I will look into the keyboard API. I wasn't seeing a way to select the text box. Once I do, the keyboard API sounds like a good solution.

**Comment by mxschmitt** [comment]
cool, let us know if it does not work for you, closing in the meantime! If you re-file with more issues, please also keep in mind that we don't know anything about ASPxHtmlEditor Class, so having a small demo with a clear repro would be really helpful.

**Comment by eatonjl** [comment]
For anyone needing help with this in the future - here is the fix my team and I found:
There was no issue with Playwright or the HTML editor we are using. But it was confusing to find out the issue it was having. 
Scenario: The HTML editor was in a nested iframe. I could write code that would detect the body of the first iframe, but not the child iframe that was in the first one. 
Problem: It turns out that the body of the HTML editor had no height because there was no text in it as of yet. This made selecting it or filling it with text impossible.
Solution: We coded it so it would select the child iframe and then used the keyboard API to begin entering text. It looks something like this:   
```
await documentOwnerPage
.frameLocator([PARENT IFRAME LOCATOR])
.locator([CHILD IFRAME LOCATOR])
.click();
  await documentOwnerPage.keyboard.type('hello');
```
